README.md
# MotorCortex-Animetitles

**Table of Contents**

- [MotorCortex-Animetitles](#motorcortex-animetitles)
  - [Demo](#demo)
- [Intro / Features](#intro--features)
- [Getting Started](#getting-started)
  - [Installation](#installation)
  - [Importing and Loading](#importing-and-loading)
- [Creating Incidents](#creating-incidents)
  - [RotatedLine](#rotatedline)
  - [RollingText](#rollingtext)
  - [RotatedLineReveal](#rotatedlinereveal)
  - [SvgBorder](#svgborder)
  - [Circle](#circle)
  - [SvgDraw](#svgdraw)
  - [LogoBox](#logobox)
  - [CircularText](#circulartext)
  - [LetterScale](#letterscale)
  - [RightOpacity](#rightopacity)
- [Adding Incidents in your clip](#adding-incidents-in-your-clip)
- [Contributing](#contributing)
- [License](#license)
- [Sponsored by](#sponsored-by)

## Demo

[Check it out here](https://donkeyclip.github.io/motorcortex-animetitles/demo/)

# Intro / Features

Using MotorCortex-Animetitles you can animate perfect titles for your clips, via its easy to use incidents.

This Plugin exposes ten incidents:

- RotatedLine
- RollingText
- RotatedLineReveal
- SvgBorder
- Circle
- SvgDraw
- LogoBox
- CircularText
- LetterScale
- RightOpacity

# Getting Started

## Installation

```bash
$ npm install --save @donkeyclip/motorcortex-animetitles
# OR
$ yarn add @donkeyclip/motorcortex-animetitles
```

## Importing and loading

```javascript
import { loadPlugin } from "@donkeyclip/motorcortex";
import TitlesPlugin from "@donkeyclip/motorcortex-animetitles";
const Titles = loadPlugin(TitlesPlugin);
```

# Creating Incidents

## RotatedTextReveal

```javascript
const nameOfIncident = new Titles.RotatedTextReveal(
  {
    text: "Rotated Reveal Incident",
    width: 600,
    color: "black",
    fontFamily: "Righteous",
    fontSize: 40,
    stagger: "0, 300",
    exitTime: 1000,
  },
  { duration: 2000, selector: ".rotatedtextreveal" }
);
```

### RotatedTextReveal Attrs

| Name       |               Are                |                    Values |
| ---------- | :------------------------------: | ------------------------: |
| width      |  total width of clip container   |      all positive numbers |
| fontSize   |         size of the font         |      all positive numbers |
| color      |        the color of text         |         hex values or RGB |
| text       |               text               |               normal text |
| stagger    | stagger between words or letters | valid Motorcortex stagger |
| fontFamily |           font family            |                    string |
| exitTime   |    time for animation to exit    |                    number |

## TextReveal

```javascript
const nameOfIncident = new new Titles.TextReveal(
  {
    text: "Reveal Incident",
    width: 500,
    color: "black",
    fontFamily: "Righteous",
    fontSize: 40,
    stagger: "0, 300,0.5,linear,omni",
    wordSplit: false,
    exitTime: 1000,
  },
  { duration: 2500, selector: ".rollingtext" }
)();
```

### TextReveal Attrs

| Name       |               Are                |                    Values |
| ---------- | :------------------------------: | ------------------------: |
| width      |  total width of clip container   |      all positive numbers |
| fontSize   |         size of the font         |      all positive numbers |
| color      |        the color of text         |         hex values or RGB |
| text       |               text               |               normal text |
| stagger    | stagger between words or letters | valid Motorcortex stagger |
| fontFamily |           font family            |                    string |
| wordSplit  | font splitting or word splitting |                true,false |
| exitTime   |    time for animation to exit    |                    number |

## RotatedLine

```javascript
const nameOfIncident = new Titles.RotatedLine(
  {
    width: 400,
    size: "XL",
    lineColor: "#ff0000",
    textColor: "#000",
    subTitle1: "test",
    subTitle2: "test2",
    subTitle3: "test3",
    title: "dokimi",
    stopOnLast: false,
    delayEnd: 3000,
    fontFamily: "Rubik Mono One",
  },
  {
    selector: ".rotatedline",
  }
);
```

### RotateLine Attrs

| Name                              |                       Are                       |               Values |
| --------------------------------- | :---------------------------------------------: | -------------------: |
| width                             |          total width of clip container          | all positive numbers |
| size                              |            scale of clip components             |     "S","M","L","XL" |
| lineColor                         |                the color of line                |    hex values or RGB |
| textColor                         |                the color of text                |    hex values or RGB |
| subTitle1 , subTitle2 , subTitle3 |                   right texts                   |          normal text |
| title                             |                    left text                    |          normal text |
| stopOnLast                        | if you like to stop animation on last text line |           true,false |
| delayEnd                          |           delay before last animation           |          millisecond |
| fontFamily                        |                   font family                   |               string |

## RollingText

```javascript
const nameOfIncident = new Titles.RollingText(
  {
    width: 450,
    size: "XL",
    lineColor: "#ff0000",
    textColor: "#000",
    subTitle1: "The",
    subTitle2: "MotorCortex Team",
    subTitle3: "Presents",
    stopOnLast: false,
    delayEnd: 1000,
    fontFamily: "Commissioner",
  },
  {
    selector: ".rollingtext",
  }
);
```

### Rollingtext Attrs

| Name                              |                       Are                       |                        Values |
| --------------------------------- | :---------------------------------------------: | ----------------------------: |
| width                             |          total width of clip container          |          all positive numbers |
| size                              |            scale of clip components             | "S","M","L","XL","XXL","XXXL" |
| lineColor                         |                the color of line                |             hex values or RGB |
| textColor                         |                the color of text                |             hex values or RGB |
| subTitle1 , subTitle2 , subTitle3 |               rolling text lines                |                   normal text |
| stopOnLast                        | if you like to stop animation on last text line |                    true,false |
| delayEnd                          |           delay before last animation           |                   millisecond |
| fontFamily                        |                   font family                   |                        string |

## RotatedLineReveal

```javascript
const nameOfIncident = new Titles.RotatedLineReveal(
  {
    width: 800,
    size: "M",
    lineColor: "#ff0000",
    textColor: "#000",
    title: "Animetitles",
    subtitle: "MC Plugin",
    leftEnd: 100,
    stopOnLast: false,
    fontFamily: "Commissioner",
  },
  {
    selector: ".rotatedlinereveal",
  }
);
```

### RotatedLineReveal Attrs

| Name       |                            Are                            |               Values |
| ---------- | :-------------------------------------------------------: | -------------------: |
| width      |               total width of clip container               | all positive numbers |
| size       |                 scale of clip components                  |          "S","M","L" |
| lineColor  |                     the color of line                     |    hex values or RGB |
| textColor  |                     the color of text                     |    hex values or RGB |
| title      |                        title text                         |          normal text |
| subtitle   |                       subtitle text                       |          normal text |
| leftEnd    | left distance of componet from the left side of container |              numbers |
| stopOnLast |      if you like to stop animation on last text line      |           true,false |
| delayEnd   |                delay before last animation                |          millisecond |
| fontFamily |                        font family                        |               string |

## SvgBorder

```javascript
const nameOfIncident = new Titles.SvgBorder(
  {
    size: "L",
    lineColor: "#ff0000",
    textColor: "#000",
    subTitle: "Developed by",
    slogan: "-KMB-",
    title: "MotorCortex Team",
    stopOnLast: false,
    fontFamily: "Commissioner",
    borderGap: 300,
  },
  {
    selector: ".svgborder",
  }
);
```

### SvgBorder Attrs

| Name       |                       Are                       |               Values |
| ---------- | :---------------------------------------------: | -------------------: |
| width      |          total width of clip container          | all positive numbers |
| size       |            scale of clip components             |          "S","M","L" |
| lineColor  |                the color of line                |    hex values or RGB |
| textColor  |                the color of text                |    hex values or RGB |
| subTitle   |                  top left text                  |          normal text |
| slogan     |                under border text                |          normal text |
| title      |                   title text                    |          normal text |
| stopOnLast | if you like to stop animation on last text line |           true,false |
| delayEnd   |           delay before last animation           |          millisecond |
| fontFamily |                   font family                   |               string |

## Circle

```javascript
const nameOfIncident = new Titles.Circle(
  {
    fontSize: 40,
    circleColor: "#ff0000",
    textColor: "#000",
    title: "ANIMATED",
    subTitle: "TITLES",
    stopOnLast: false,
    fontFamily: "Righteous",
    width: 500,
  },
  {
    selector: ".circle",
  }
);
```

#### IMPORTANT

In order to use tin incident you should include the img.svg that is in demo folder in your clip.js directory

### Circle Attrs

| Name        |                             Are                              |               Values |
| ----------- | :----------------------------------------------------------: | -------------------: |
| fontSize    |                       size of the font                       | all positive numbers |
| width       | total width of clip container and width/height of the circle | all positive numbers |
| circleColor |                     the color of circle                      |    hex values or RGB |
| textColor   |                      the color of text                       |    hex values or RGB |
| subTitle    |                        top left text                         |          normal text |
| title       |                svg cobe with 1 path tag only                 |             svg text |
| stopOnLast  |       if you like to stop animation on last text line        |           true,false |
| delayEnd    |                 delay before last animation                  |          millisecond |
| fontFamily  |                         font family                          |               string |

## LogoBox

```javascript
const nameOfIncident = new Titles.LogoBox(
  {
    size: 2,
    textColor: "#fff",
    bgColor: "#000",
    subTitle1: "testdawdawddawd",
    subTitle2: "test2",
    subTitle3: "test3",
    width: 200,
    stopOnLast: false,
    logoUrl:
      "https://donkeyclip.github.io/motorcortex-animetitles/demo/logo.png",
    fontFamily: "Rubik Mono One",
  },
  {
    selector: ".logobox",
  }
);
```

#### IMPORTANT

In order to use tin incident you should include the logo.png image in your clip.js directory

### LogoBox Attrs

| Name                              |                       Are                       |               Values |
| --------------------------------- | :---------------------------------------------: | -------------------: |
| size                              |           change the size of element            | all positive numbers |
| bgColor                           |                 the color of bg                 |    hex values or RGB |
| textColor                         |                the color of text                |    hex values or RGB |
| subTitle1 , subTitle2 , subTitle3 |                   right texts                   |          normal text |
| width                             |          total width of text container          | all positive numbers |
| stopOnLast                        | if you like to stop animation on last text line |           true,false |
| logoUrl                           |                 url of an image                 |               string |
| fontFamily                        |                   font family                   |               string |

## CircularText

```javascript
const nameOfIncident = new Titles.CircularText(
  {
    width: 500,
    height: 500,
    color: "#ff0000",
    text: "CircularText - AnimeTitles - MotorCortex - Plugin -",
    fontSize: 10,
    viewBox: 100,
    path: 38,
    fill: "#ff000000",
    fontFamily: "Righteous",
  },
  {
    selector: ".circulartext",
    repeats: 2,
  }
);
```

### CircularText Attrs

| Name       |                 Are                 |            Values |
| ---------- | :---------------------------------: | ----------------: |
| width      |    total width of clip container    |            number |
| height     |   total height of clip container    |            number |
| color      |         color for the text          | hex values or RGB |
| text       |         the displayed text          |            string |
| fontSize   |          size of the font           |            number |
| fontFamily |             font family             |            string |
| viewBox    | width for the wrapper of the circle |            number |
| path       |    width for the circle's radius    |            number |
| fill       |       color for the svg shape       | hex values or RGB |

## LetterScale

```javascript
const nameOfIncident = new Titles.LetterScale(
  {
    width: 900,
    height: 900,
    color: "rebeccapurple",
    text: "LetterScale",
    fontSize: 100,
    stroke: false,
    strokeSize: 2,
    strokeColor: "rebeccapurple",
    fontFamily: "Righteous",
  },
  {
    selector: ".letterscale",
  }
);
```

### LetterScale Attrs

| Name        |              Are               |            Values |
| ----------- | :----------------------------: | ----------------: |
| width       | total width of clip container  |            number |
| height      | total height of clip container |            number |
| color       |       color for the text       | hex values or RGB |
| text        |       the displayed text       |            string |
| fontSize    |        size of the font        |            number |
| fontFamily  |          font family           |            string |
| stroke      |     a border to SVG shapes     |           boolean |
| strokeSize  |       the stroke's size        |            number |
| strokeColor |       the stroke's color       | hex values or RGB |

## RightOpacity

```javascript
const nameOfIncident = new Titles.RightOpacity(
  {
    width: 1000,
    height: 250,
    text: "RightOpacity",
    color: "#0000",
    fontSize: 100,
    stroke: true,
    strokeSize: 2,
    strokeColor: "rebeccapurple",
    fontFamily: "Rubik Mono One",
    exitTime: 1000,
  },
  {
    selector: ".rightopacity",
  }
);
```

### RightOpacity Attrs

| Name        |              Are               |            Values |
| ----------- | :----------------------------: | ----------------: |
| width       | total width of clip container  |            number |
| height      | total height of clip container |            number |
| color       |       color for the text       | hex values or RGB |
| text        |       the displayed text       |            string |
| fontSize    |        size of the font        |            number |
| fontFamily  |          font family           |            string |
| stroke      |     a border to SVG shapes     |           boolean |
| strokeSize  |       the stroke's size        |            number |
| strokeColor |       the stroke's color       | hex values or RGB |
| exitTime    |   time for animation to exit   |            number |

# Adding Incidents in your clip

```javascript
clipName.addIncident(incidentName, startTime);
```

# Contributing

In general, we follow the "fork-and-pull" Git workflow, so if you want to submit patches and additions you should follow the next steps:

1. **Fork** the repo on GitHub
2. **Clone** the project to your own machine
3. **Commit** changes to your own branch
4. **Push** your work back up to your fork
5. Submit a **Pull request** so that we can review your changes

# License

[MIT License](https://opensource.org/licenses/MIT)

# Sponsored by

[<img src="https://presskit.donkeyclip.com/logos/donkey%20clip%20logo.svg" width=250></img>](https://donkeyclip.com)


CHANGELOG.md
# Changelog

## [1.4.0](https://github.com/donkeyclip/motorcortex-animetitles/compare/v1.3.0...v1.4.0) (2022-12-07)


### Features

* add text tranform ([7c8ed51](https://github.com/donkeyclip/motorcortex-animetitles/commit/7c8ed516984d57b4885c8e6a8a2ba6f12d652dc2))

## [1.3.0](https://github.com/donkeyclip/motorcortex-animetitles/compare/v1.2.3...v1.3.0) (2022-10-04)


### Features

* add new incidents and clear readme ([6786def](https://github.com/donkeyclip/motorcortex-animetitles/commit/6786def3a3dcb34da8a7ef565c0746e0c7ea8098))
* new incidents ([c3648bf](https://github.com/donkeyclip/motorcortex-animetitles/commit/c3648bf835f77be6862a7d35cf96fe7ec712baf3))

### [1.2.3](https://github.com/donkeyclip/motorcortex-animetitles/compare/v1.2.2...v1.2.3) (2022-09-29)


### Bug Fixes

* update dependencies ([db2d520](https://github.com/donkeyclip/motorcortex-animetitles/commit/db2d5204c4b19d1a3637020b39076c1f16ca5b0a))

### [1.2.2](https://github.com/donkeyclip/motorcortex-animetitles/compare/v1.2.1...v1.2.2) (2022-09-29)


### Bug Fixes

* add font family to validation and readme ([f7eee0d](https://github.com/donkeyclip/motorcortex-animetitles/commit/f7eee0d986caab14e9e82663d029098f149b6c2a))

### [1.2.1](https://github.com/donkeyclip/motorcortex-animetitles/compare/v1.2.0...v1.2.1) (2022-04-11)


### Bug Fixes

* delete commitlint.config.js ([16bf946](https://github.com/donkeyclip/motorcortex-animetitles/commit/16bf94696cd2a4f5d7aed07e1f11ecbb54863e00))
* rename RollingText and RotatedLineReveal Incidents ([bab01f6](https://github.com/donkeyclip/motorcortex-animetitles/commit/bab01f6d42390126842aec1694c153ca0802aa6d))

## [1.2.0](https://github.com/donkeyclip/motorcortex-animetitles/compare/v1.1.0...v1.2.0) (2022-04-07)


### Features

* drop explorer 11 support ([d17ea56](https://github.com/donkeyclip/motorcortex-animetitles/commit/d17ea56f0a129ec0bfc2a3b4cd063ae7d5d65ce8))
* update compatibility with motorcortex ([ad93113](https://github.com/donkeyclip/motorcortex-animetitles/commit/ad931139b2967bee065a661fa7f781489a3e45cb))

## [1.1.0](https://www.github.com/donkeyclip/motorcortex-animetitles/compare/v1.0.6...v1.1.0) (2022-03-17)


### Features

* declare mc 8 as compatible ([d4c3851](https://www.github.com/donkeyclip/motorcortex-animetitles/commit/d4c38517939a74180959c8874f1b306ed88a1384))

### [1.0.6](https://www.github.com/donkeyclip/motorcortex-animetitles/compare/v1.0.5...v1.0.6) (2022-03-01)


### Bug Fixes

* fix circleanimation and safari bugs ([aea761a](https://www.github.com/donkeyclip/motorcortex-animetitles/commit/aea761ab1657c45eeb4ab4f35d8316b83bfe3522))

### [1.0.5](https://www.github.com/donkeyclip/motorcortex-animetitles/compare/v1.0.4...v1.0.5) (2022-01-31)


### Bug Fixes

* remove anime plugin ([4d0a00c](https://www.github.com/donkeyclip/motorcortex-animetitles/commit/4d0a00c05d83d2dc90666b2f20257c1ae4aea7c7))

### [1.0.4](https://www.github.com/donkeyclip/motorcortex-animetitles/compare/v1.0.3...v1.0.4) (2022-01-30)


### Bug Fixes

* drop code from demo config ([6147d09](https://www.github.com/donkeyclip/motorcortex-animetitles/commit/6147d091a741697c006138cbd547df3a8cfb1346))
* fix broken rollup config ([81d56da](https://www.github.com/donkeyclip/motorcortex-animetitles/commit/81d56da5779e8d5d0d3ecea20854485b34494dce))
* fix broken webpack config ([1f92033](https://www.github.com/donkeyclip/motorcortex-animetitles/commit/1f92033ab32c45f1f11be2ee85fce99209ea48c3))
* update build and demo files drop old babel config ([fdd1449](https://www.github.com/donkeyclip/motorcortex-animetitles/commit/fdd144995e557c03ce3fd75feef67b67efacb3a3))
* update build and demo files drop old babel config ([0eaef94](https://www.github.com/donkeyclip/motorcortex-animetitles/commit/0eaef941790acf89e3eb7f22794d1ac93d1a2acf))
* update build and demo files drop old babel config ([bfc7e2a](https://www.github.com/donkeyclip/motorcortex-animetitles/commit/bfc7e2a20a53e1dd76210f1b9bb8a998a1952797))
* update build and demo files drop old babel config ([b3f992f](https://www.github.com/donkeyclip/motorcortex-animetitles/commit/b3f992fe49ba5c9b1381dab845135f728ee65d12))
* update build and demo files drop old babel config ([03131d5](https://www.github.com/donkeyclip/motorcortex-animetitles/commit/03131d577d7e0e351b93ab03c05351c3363fadfe))
* update build and demo files drop old babel config ([d59d064](https://www.github.com/donkeyclip/motorcortex-animetitles/commit/d59d0645b6d1301787bcdb94bbb83abf9ae7db71))
* update build and demo files drop old babel config ([e7a1e17](https://www.github.com/donkeyclip/motorcortex-animetitles/commit/e7a1e1766b5cdc11d462ea3c553186dad87a9584))
* update rollup deps to latest packages ([693e426](https://www.github.com/donkeyclip/motorcortex-animetitles/commit/693e42659d824ff82ff8d9a508421e3acebc13db))

### [1.0.3](https://www.github.com/donkeyclip/motorcortex-animetitles/compare/v1.0.2...v1.0.3) (2021-09-30)


### Bug Fixes

* lint ([3d6ccad](https://www.github.com/donkeyclip/motorcortex-animetitles/commit/3d6ccade3c062a2311287e39a0857bd90fd49a44))
* remove timing attribute from incidents ([2a9cad8](https://www.github.com/donkeyclip/motorcortex-animetitles/commit/2a9cad84b02d4c3b2cdb5d49941eeae9cd69a5fe))
* update deps ([8de61ae](https://www.github.com/donkeyclip/motorcortex-animetitles/commit/8de61ae2b2acb510aa17fec72231ccfd069509a5))
* update npm name ([b0a4813](https://www.github.com/donkeyclip/motorcortex-animetitles/commit/b0a4813e97535fe9fe184459237c9f2fd5785a21))

### 1.0.2 (2021-04-16)


### Features

* add delay ([8fd8f16](https://www.github.com/kissmybutton/motorcortex-animetitles/commit/8fd8f1678cf5356ba134e69c42b92ef0e5195e56))
* new mc and player version ([cf826ee](https://www.github.com/kissmybutton/motorcortex-animetitles/commit/cf826ee5bd93103b8648e73383813f03f94a54e0))
* update dependencies ([2f8b3c7](https://www.github.com/kissmybutton/motorcortex-animetitles/commit/2f8b3c786a3c020c5328cf8b4f7098967f6cb6d8))


### Bug Fixes


demo/index.js
import { HTMLClip, loadPlugin, CSSEffect } from "@donkeyclip/motorcortex";
import Player from "@donkeyclip/motorcortex-player";
import TitlesPlugin from "../dist/motorcortex-animetitles.esm";

const Titles = loadPlugin(TitlesPlugin);

const clip = new HTMLClip({
  css: `
    body{
      background-color : white;
    }

    .container{
      position: relative;
      background:linear-gradient(141deg, #ccc 25%, #eee 40%, #ddd 55%);
      height:100%;
      display: flex;
      justify-content: space-around;
      align-items: center;
      flex-direction: column;
    }

    .row{
      display: flex;
      justify-content: space-around;
      align-items: center;
    }

    .cel{
      display: flex;
      justify-content: center;
      align-items: center;
      flex-direction: column;
      position: absolute;
    }

    .full{
      display: flex;
      justify-content: center;
      align-items: center;
    }
  `,
  html: `
    <div class="container">
      <div class="row" >
        <div class="cel"> <div class="textreveal full"></div>  </div>
        <div class="cel"> <div class="rotatedtextreveal full"></div>  </div>
        <div class="cel"> <div class="rollingtext full"></div>  </div>
        <div class="cel"><div class="svgborder full"></div> </div>
        <div class="cel"> <div class="rotatedline full"></div> </div>
        <div class="cel"> <div class="rotatedlinereveal full"></div> </div>
        <div class="cel"><div class="svgdraw full"></div> </div>
        <div class="cel"><div class="circle full"></div> </div>
        <div class="cel"><div class="logobox full"></div> </div>
        <div class="cel"><div class="textwriting full"></div> </div>
        <div class="cel"><div class="rightopacity full"></div> </div>
        <div class="cel"><div class="letterscale full"></div> </div>
        <div class="cel"><div class="circulartext full"></div> </div>
      </div>
    </div>
  `,
  fonts: [
    {
      type: `google-font`,
      src: `https://fonts.googleapis.com/css2?family=Rubik+Mono+One&display=swap`,
    },
    {
      type: `google-font`,
      src: `https://fonts.googleapis.com/css2?family=Righteous&display=swap`,
    },
    {
      type: "google-font",
      src: "https://fonts.googleapis.com/css2?family=Commissioner:wght@100;200;300;400;500;600;700;800;900&display=swap",
    },
  ],
  host: document.getElementById("clip"),
  containerParams: {
    width: "1280px",
    height: "720px",
  },
});

const rolinText = new Titles.RollingText(
  {
    width: 450,
    size: "XL",
    lineColor: "#ff0000",
    textColor: "#000",
    subTitle1: "The",
    subTitle2: "MotorCortex Team",
    subTitle3: "Presents",
    stopOnLast: false,
    delayEnd: 1000,
    fontFamily: "Commissioner",
  },
  {
    selector: ".rollingtext",
  }
);

const rotatedlinereveal = new Titles.RotatedLineReveal(
  {
    width: 800,
    size: "M",
    lineColor: "#ff0000",
    textColor: "#000",
    title: "Animetitles",
    subtitle: "MC Plugin",
    leftEnd: 100,
    stopOnLast: false,
    fontFamily: "Commissioner",
  },
  {
    selector: ".rotatedlinereveal",
  }
);

const rotatedline = new Titles.RotatedLine(
  {
    width: 400,
    size: "XL",
    lineColor: "#ff0000",
    textColor: "#000",
    subTitle1: "Minimal",
    subTitle2: "Elegant",
    subTitle3: "Modern",
    title: "Title",
    stopOnLast: false,
    fontFamily: "Commissioner",
  },
  {
    selector: ".rotatedline",
  }
);

const svgborder = new Titles.SvgBorder(
  {
    size: "L",
    lineColor: "#ff0000",
    textColor: "#000",
    subTitle: "Developed by",
    slogan: "-KMB-",
    title: "MotorCortex Team",
    stopOnLast: false,
    fontFamily: "Commissioner",
    borderGap: 300,
  },
  {
    selector: ".svgborder",
  }
);

const circle = new Titles.Circle(
  {
    fontSize: 70,
    circleColor: "#ff0000",
    textColor: "#000",
    title: "ANIMATED",
    subTitle: "TITLES",
    stopOnLast: false,
    fontFamily: "Righteous",
    width: 500,
  },
  {
    selector: ".circle",
  }
);

const logobox = new Titles.LogoBox(
  {
    size: 2,
    textColor: "#fff",
    bgColor: "#000",
    subTitle1: "TYPOGRAPHY",
    subTitle2: "ANIMATION",
    subTitle3: "BY",
    width: 200,
    stopOnLast: false,
    fontFamily: "Commissioner",
    logoUrl:
      "https://donkeyclip.github.io/motorcortex-animetitles/demo/assets/logo.png",
  },
  {
    selector: ".logobox",
  }
);

const RightOpacity = new Titles.RightOpacity(
  {
    width: 1000,
    height: 250,
    text: "RightOpacity",
    color: "#0000",
    fontSize: 100,
    stroke: true,
    strokeSize: 2,
    strokeColor: "rebeccapurple",
    fontFamily: "Rubik Mono One",
    exitTime: 1000,
  },
  {
    duration: 3000,
    selector: ".rightopacity",
  }
);

const rightopacityOpacity = new CSSEffect(
  {
    animatedAttrs: {
      opacity: 1,
    },
    initialValues: {
      opacity: 0,
    },
  },
  {
    duration: 1,
    selector: ".rightopacity",
  }
);

const LetterScale = new Titles.LetterScale(
  {
    width: 900,
    height: 900,
    color: "rebeccapurple",
    text: "LetterScale",
    fontSize: 100,
    stroke: false,
    strokeSize: 2,
    strokeColor: "rebeccapurple",
    fontFamily: "Righteous",
  },
  {
    selector: ".letterscale",
  }
);

const LetterScaleOpacity = new CSSEffect(
  {
    animatedAttrs: {
      opacity: 1,
    },
    initialValues: {
      opacity: 0,
    },
  },
  {
    duration: 1,
    selector: ".letterscale",
  }
);
const LetterScaleOpacityBack = new CSSEffect(
  {
    animatedAttrs: {
      opacity: 0,
    },
  },
  {
    duration: 1,
    selector: ".letterscale",
  }
);

const CircularText = new Titles.CircularText(
  {
    width: 500,
    height: 500,
    color: "#ff0000",
    text: "CircularText - AnimeTitles - MotorCortex - Plugin -",
    fontSize: 10,
    viewBox: 100,
    path: 38,
    fill: "#ff000000",
    fontFamily: "Righteous",
  },
  {
    selector: ".circulartext",
    repeats: 2,
  }
);

const CircularTextOpacity = new CSSEffect(
  {
    animatedAttrs: {
      opacity: 1,
    },
    initialValues: {
      opacity: 0,
    },
  },
  {
    duration: 1,
    selector: ".circulartext",
  }
);

const TextRevealIncident = new Titles.TextReveal(
  {
    text: "Reveal Incident",
    width: 500,
    color: "black",
    fontFamily: "Righteous",
    fontSize: 40,
    stagger: "0, 300,0.5,linear,omni",
    exit: "bottom",
    textTransform: "none",
  },
  { duration: 2000, selector: ".textreveal" }
);
const RotatedTextRevealIncident = new Titles.RotatedTextReveal(
  {
    text: "Rotated Reveal Incident",
    width: 600,
    color: "black",
    fontFamily: "Righteous",
    fontSize: 40,
    stagger: "0, 300",
    exitTime: 1000,
    textTransform: "none",
  },
  { duration: 2000, selector: ".rotatedtextreveal" }
);

clip.addIncident(TextRevealIncident, 0);
clip.addIncident(RotatedTextRevealIncident, clip.calculatedDuration);
clip.addIncident(rolinText, clip.calculatedDuration);
clip.addIncident(rotatedlinereveal, clip.calculatedDuration);
clip.addIncident(svgborder, clip.calculatedDuration);
clip.addIncident(rotatedline, clip.calculatedDuration);
clip.addIncident(circle, clip.calculatedDuration);
clip.addIncident(logobox, clip.calculatedDuration);
clip.addIncident(rightopacityOpacity, clip.calculatedDuration);
clip.addIncident(RightOpacity, clip.calculatedDuration);
clip.addIncident(LetterScaleOpacity, clip.calculatedDuration);
clip.addIncident(LetterScale, clip.calculatedDuration);
clip.addIncident(LetterScaleOpacityBack, clip.calculatedDuration);
clip.addIncident(CircularTextOpacity, clip.calculatedDuration);
clip.addIncident(CircularText, clip.calculatedDuration);

new Player({ clip });
